lib/spack/spack/version.py

```diff
@@ -587,14 +604,14 @@ def _cmp(self, other_lookups=None):
 
         return self.version
 
+    @coerced
+    def overlaps(self, other):
+        return self.satisfies(other) or other.satisfies(self)
```

Comment:
Same here, I personally find it confusing because `x ∩ y ≠ Ø ⟺  (x ⊂ y) ∨ (x ⊃ y)` is false

Reply:
Yes, you're correct in general. I think it's true for `VersionBase` objects but I can change that to be more explicit.

Reply:
Done in https://github.com/spack/spack/pull/35681/commits/76d4ef974669290b9d8026f8bb4fbf29c292dff2